controllers/platformoperator_controller.go

```diff
@@ -102,10 +109,51 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	u := updater.NewPlatformOperatorUpdater(r.Client)
+	defer func() {
+		if err := u.Apply(ctx, po); err != nil {
+			log.Error(err, "failed to update status")
+		}
+	}()
+
+	u.UpdateStatus(
+		updater.SetPhase(platformopenshiftiov1alpha1.PhaseFailing),
+		updater.EnsureCondition(metav1.Condition{
+			Type:    platformopenshiftiov1alpha1.TypeInstalled,
+			Status:  metav1.ConditionUnknown,
+			Reason:  platformopenshiftiov1alpha1.ReasonInstalling,
+			Message: "",
+		}),
+	)
+
+	rc, err := handleCatalogConnection(catalogSourceAddress)
```

Comment:
this function is called `handleCatalogConnection` but returns a registry client, which is a bit misleading. Maybe something like `newRegistryClient()` is more appropriate?

Reply:
Yeah, I was going back and fourth on this name since I was trying to determine how exactly I would propagate this connection. This is ultimately where I settled but I agree with you that `newRegistryClient` or something similar fits better.